SCons/CacheDir.py

```diff
@@ -209,22 +209,28 @@ def _mkdir_atomic(self, path: str) -> bool:
         if os.path.exists(directory):
             return False
 
+        # TODO: tried to use TemporaryDirectory() here and the result as a
+        #   context manager, but that fails on Python 3.7 (works on 3.8+) after
+        #   the directory has successfully been renamed. Not sure why.
```

Comment:
Maybe leave the old implementation commented out with `#TODO dropping python3.7` or somesuch

Reply:
can do that... I didn't because several lines long due to indent change for no longer using context mgr